use config to drive autoscroll state

movalex opened the pull request.

m0rg5 commented:
This behaviour persists in latest 2.10.3, mac m1 Sonoma 14.2.1

movalex commented:
> This behaviour persists in latest 2.10.3, mac m1 Sonoma 14.2.1

Yep, this will persist until this PR is merged or until the authors implement their own scroll controller.
If you need that right now, feel free to try and test it in my fork.
https://github.com/movalex/ChatGPT-Next-Web.git

H0llyW00dzZ commented:
tbh this pretty useful but, I don't think so this pr will be merged, since the owner of this repository has changed.

H0llyW00dzZ commented on app/components/chat.tsx:
```diff
@@ -379,12 +379,13 @@ function useScrollToBottom() {
   function scrollDomToBottom() {
     const dom = scrollRef.current;
     if (dom) {
       requestAnimationFrame(() => {
```
Just info 
this `requestAnimationFrame` stupid complex that made performance in browser bad when there is lots of messages

movalex replied:
I don't know this repo much, but I think they've added this recently to make the show text animation smoother. If this impacts performance so bad, I guess it should be either an option or omitted.

H0llyW00dzZ replied:
I was looking around it yesterday trying to Improve it

H0llyW00dzZ reviewed (commented):
Just info

H0llyW00dzZ commented:
> > tbh this pretty useful but, I don't think so this pr will be merged, since the owner of this repository has changed.
> 
> Why does the owner change would prevent them from merging it?

I have no idea

movalex commented:
Btw you did a great work with your fork. I like the search bar in the chat list!
I added a few changes to my repo recently too. I use Upstash to sync the chats, so I added a Force Push option to replace data on remote. 
https://github.com/movalex/ChatGPT-Next-Web/tree/feat/add-force-sync-Upstash-option

H0llyW00dzZ commented:
> Btw you did a great work with your fork. I like the search bar in the chat list! I added a few changes to my repo recently too. I use Upstash to sync the chats, so I added a Force Push option to replace data on remote. https://github.com/movalex/ChatGPT-Next-Web/tree/feat/add-force-sync-Upstash-option

it was created by `Andision` about search bar. Also, I might add my own chat synchronization feature later, which will be written in `Go`.

m0rg5 commented:
> > This behaviour persists in latest 2.10.3, mac m1 Sonoma 14.2.1
> 
> Yep, this will persist until this PR is merged or until the authors implement their own scroll controller. If you need that right now, feel free to try and test it in my fork. https://github.com/movalex/ChatGPT-Next-Web/tree/develop

Can you clarify/link to how? Im asking CGPT and getting a bit lost

H0llyW00dzZ commented:
@movalex I've cherry-picked your PR into my fork. It's quite useful.
